### PR TITLE
Add more conditions for deletion; add more configuration; add Travis image setup for GPR

### PR DESCRIPTION
**Pull request by tremaineeto**
- Add deletion over a certain deployment date
- Add excludable namespaces
- Add deletion of HPAs, Ingresses, and Services
- Add Boolean values to opt-in or opt-out of particular resource deletion
- Add GitHub Package Registry set-up in Travis YAML

### CODE REVIEW

**Comment by tremaineeto** [comment]
I'm going to address the Travis CI build failing as I try to get the image creation and push working with GitHub Package Registry, but the code is ready for review otherwise.

**Comment by tremaineeto** [comment]
@bhaskar-reddy @timjonesdev et al., I was able to get the Travis checks to pass on this PR.

File: .travis.yml [hunk 1]
```diff
@@ -5,3 +5,6 @@ go:
 
 script:
   - env GO111MODULE=on go build
+  - echo "$DOCKER_PASSWORD" | docker login docker.pkg.github.com --username "$DOCKER_USERNAME" --password-stdin
```
Comment (commissionerjgordon) [inline .travis.yml]: This probably is just feeding the password value into the second command, but can we double check the echo does not log the password to the console?

File: config.go [hunk 2]
```diff
@@ -45,29 +51,33 @@ func (dc *KleanerConfig) SetFunctions(fnMap map[string]DeleteFunc) {
 }
 
 func init() {
+	v := viper.New()
```
Comment (commissionerjgordon) [inline config.go:54]: since I'm new to this project. What does viper do?

File: .travis.yml (hunk 1 above)
Comment (tremaineeto) [reply .travis.yml]: Yeah, when I looked at Travis logs, the password isn't logged. However, there are still improvements we can make by figuring out how to use a credential store.

File: config.go (hunk 2 above)
Comment (tremaineeto) [reply config.go:54]: [Viper](https://github.com/spf13/viper) is a project that basically allows for Go configuration.

**Comment by tremaineeto** [comment]
Seems to me that the whole credentials store thing in the Travis file is misleading, at least according to this [closed issue](https://github.com/concourse/docker-image-resource/issues/243).

Anyway, I've got the build working again.

**Pull request merged**